performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/UITests.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Copyright 2021 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+
+class UITests: XCTestCase {
+  var app: XCUIApplication = XCUIApplication()
+
+  override func setUpWithError() throws {
+    #if !os(iOS) && !os(tvOS)
+      throw Error("Unsupported platform.")
+    #endif
+
+    continueAfterFailure = false
+
+    app.launch()
+  }
+
+  override func tearDownWithError() throws {
+    app.terminate()
+  }
+
+  func checkButtons(_ functions: [String], timeout: TimeInterval = 1) throws {
+    for function in functions {
+      let button = app.buttons["\(function) Image"]
+      XCTAssert(button.waitForExistence(timeout: timeout), "Missing \(function) button.")
+      XCTAssert(button.isEnabled, "\(function) button not enabled.")
+      XCTAssert(button.isHittable, "\(function) button not in view.")
+    }
+  }
+
+  func checkText(_ text: String, timeout: TimeInterval = 1) throws {
+    let element = app.staticTexts[text]
+    XCTAssert(element.waitForExistence(timeout: timeout), "Missing text '\(text)'.")
+    XCTAssert(element.isHittable, "Text '\(text)' not in view.")
+  }
+
+  func checkStatus(_ status: String, timeout: TimeInterval = 1) throws {
+    #if os(iOS)
+      try checkText(status, timeout: timeout)
+    #endif
+  }
```

Comment:
nit: you can wrap this whole function in the `#if` so it doesn't show up in the sidebar for non-iOS test runs.

Reply:
I'm not sure which sidebar you mean, but if you mean the "Test navigator" with the tests that run then it doesn't even show up since it is not an independent test (because the function name doesn't start with `test`):

![image](https://user-images.githubusercontent.com/8367227/129105422-ef4f101b-78f0-453f-a607-02b9c5cb64df.png)

Also, if I were to do this then I would have to do the same `#if` wrapping every time I call it within `testAllViews`, which is definitely not optimal. Please let me know if you think we should look for a different approach and not keep it as is.

Reply:
rationale LGTM!